### ISSUE TITLE
Webhook validation for Topology NodeDeletionTimeout and NodeDrainTimeout

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes-sigs/cluster-api#7104]
NodeDeletionTimeout and NodeDrainTimeout were added to Topology managed clusters in #7098 and #6278. Currently the values of these fields are not validated on creation, and validation is instead done when the templates are turned into objects. 

This lack of up-front validation lead to the unexpected failure in #7047. We could do some basic validation in the webhook on object creation to ensure these values are correctly formatted and in a given range before creation.

/kind feature

**Comment by sbueringer** [comment]
What would be the valid range for those fields?

**Comment by killianmuldoon** [comment]
We don't have these defined right now in the machine webhook (and I don't know if there's any need to), but defining a min/max is an optional part of this.

I think the main part is to ensure that we do enough validation to catch errors like #7047 on object creation, instead of during the reconcile.

**Comment by sbueringer** [comment]
Yup. The problem is that `metav1.Duration` just has type "string" as OpenAPI schema, right? 

If it would also use `format` `duration` OpenAPI would probably handle it for us?

**Comment by JoelSpeed** [comment]
What happens to the existing users who have persisted bad values when we update the validation here? Has it been considered to use ratcheting validation at all?

**Comment by sbueringer** [comment]
I think it was not considered

**Comment by JoelSpeed** [comment]
Ratcheting validation exists directly within the API server from Kube 1.30, but since we need to support older versions, ratcheting can either be implemented in a webhook, or, within a couple of well crafted CEL transition rules (though these aren't perfect as they don't cover the create case).

Without ratcheting, this does have the potential to break users on upgrade, they wouldn't be able to write anything to the object until the values of these broken fields were fixed.

**Comment by sbueringer** [comment]
> Ratcheting validation exists directly within the API server from Kube 1.30

If it's enabled per default it could be okay to just wait until 1.30 is the min supported version (Cluster API v1.10, basically we could then merge in December)

**Comment by sbueringer** [comment]
All of these fields have been changed to *Seconds of type *int32 in v1beta2 (with Minimum OpenAPI validation)

/close
